core/capabilities/remote/executable/parallel_executor.go

```diff
@@ -0,0 +1,58 @@
+package executable
+
+import (
+	"context"
+	"sync"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+)
+
+type task struct {
+	fnCtx *context.Context
+	fn    func(ctx context.Context)
+}
+
+type parallelExecutor struct {
+	tasksCh  chan task
+	wg       sync.WaitGroup
+	stopChan services.StopChan
+}
+
+func newParallelExecutor(maxParallelTasks int) *parallelExecutor {
+	executor := &parallelExecutor{
+		tasksCh:  make(chan task),
+		stopChan: make(services.StopChan),
+		wg:       sync.WaitGroup{},
+	}
+
+	for i := 0; i < maxParallelTasks; i++ {
+		executor.wg.Add(1)
+		go func() {
```

Comment:
Is there a benefit to having all those goroutines constantly running instead of spawning them on demand? I feel that you could get rid of the Executor object and simply spawn new goroutines in place, using another channel as a semaphore to limit the number.

Reply:
there's no benefit, but also being golang no disbenefit and gives a simple solution to the requirement to limit the number whilst waiting on the callers context should the limit be reached.  Can you expand on your suggestion?

Reply:
ok, I think I understand your suggestion, I've updated the code, but kept the logic in the parallel executor type to make it easier to test in isolation.